cnd/src/network.rs

```diff
@@ -87,18 +92,195 @@ impl<TSubstream, B> ComitNode<TSubstream, B> {
+
+                                Ok(swap_id)
+                            }
+                            (alpha_ledger, beta_ledger, alpha_asset, beta_asset) => {
+                                log::warn!(
+                                    "swapping {:?} to {:?} from {:?} to {:?} is currently not supported", alpha_asset, beta_asset, alpha_ledger, beta_ledger
+                                );
+
+                                let decline_body = DeclineResponseBody {
+                                    reason: Some(SwapDeclineReason::UnsupportedSwap),
+                                };
+
+                                Err(Box::new(futures::future::ok(
+                                    Response::empty()
+                                        .with_header(
+                                            "decision",
+                                            Decision::Declined
+                                                .to_header()
+                                                .expect("Decision should not fail to serialize"),
+                                        )
+                                        .with_body(serde_json::to_value(decline_body).expect(
+                                            "decline body should always serialize into serde_json::Value",
+                                        )),
+                                )))
+                            }
+                        }
+                    }
+                    SwapProtocol::Unknown(protocol) => {
+                        log::warn!("the swap protocol {} is currently not supported", protocol);
+
+                        let decline_body = DeclineResponseBody {
+                            reason: Some(SwapDeclineReason::UnsupportedProtocol),
+                        };
+                        Err(Box::new(futures::future::ok(
+                            Response::empty()
+                                .with_header(
+                                    "decision",
+                                    Decision::Declined
+                                        .to_header()
+                                        .expect("Decision should not fail to serialize"),
+                                )
+                                .with_body(serde_json::to_value(decline_body).expect(
+                                    "decline body should always serialize into serde_json::Value",
+                                )),
+                        )))
+                    }
+                }
+            }
+
+            // This case is just catered for, because of rust. It can only happen
+            // if there is a typo in the request_type within the program. The request
+            // type is checked on the messaging layer and will be handled there if
+            // an unknown request_type is passed in.
+            request_type => {
+                log::warn!("request type '{}' is unknown", request_type);
```

Comment:
A `warn` is a bit light if this is not supposed to happen at all.

Reply:
This used to be there before, we tried to minimize the impact of this PR :)
Happy to accept a follow-up PR on this!